apps/server/src/shared/domain/entity/task.entity.ts

```diff
@@ -99,102 +104,153 @@ export class Task extends BaseEntityWithTimestamps implements ILearnroomElement,
 		}
 		if (this.availableDate && this.availableDate > new Date(Date.now())) {
 			return true;
 		}
+
 		return false;
 	}
 
-	areSubmissionsPublic(): boolean {
-		return !!this.publicSubmissions;
+	private getSubmittedSubmissions(): Submission[] {
+		const submissions = this.getSubmissionItems();
+		const gradedSubmissions = submissions.filter((submission) => submission.isSubmitted());
+
+		return gradedSubmissions;
 	}
 
-	private getSubmissionItems(): Submission[] {
-		if (!this.submissions.isInitialized(true)) {
-			throw new Error('Submissions items are not loaded.');
-		}
-		const submissions = this.submissions.getItems();
-		return submissions;
+	public areSubmissionsPublic(): boolean {
+		const areSubmissionsPublic = !!this.publicSubmissions;
+
+		return areSubmissionsPublic;
 	}
 
-	getSubmittedUserIds(): EntityId[] {
-		const submittedUserIds = this.getSubmissionItems().map((submission) => submission.student.id);
-		const uniqueSubmittedUserIds = [...new Set(submittedUserIds)];
+	private getGradedSubmissions(): Submission[] {
+		const submissions = this.getSubmissionItems();
+		const gradedSubmissions = submissions.filter((submission) => submission.isGraded());
 
-		return uniqueSubmittedUserIds;
+		return gradedSubmissions;
 	}
 
-	getNumberOfSubmittedUsers(): number {
-		const submittedUserIds = this.getSubmittedUserIds();
-		const count = submittedUserIds.length;
+	private isSubmittedForUser(user: User): boolean {
+		const submissions = this.getSubmissionItems();
+		const isSubmitted = submissions.some((submission) => submission.isSubmittedForUser(user));
 
-		return count;
+		return isSubmitted;
 	}
 
-	getGradedUserIds(): EntityId[] {
-		const gradedUserIds = this.getSubmissionItems()
-			.filter((submission) => submission.isGraded())
-			.map((submission) => submission.student.id);
-		const uniqueGradedUserIds = [...new Set(gradedUserIds)];
+	private isGradedForUser(user: User): boolean {
+		const submissions = this.getSubmissionItems();
+		const graded = submissions.some((submission) => submission.isGradedForUser(user));
 
-		return uniqueGradedUserIds;
+		return graded;
 	}
 
-	getNumberOfGradedUsers(): number {
-		const gradedUserIds = this.getGradedUserIds();
-		const count = gradedUserIds.length;
+	private getUniqueSubmitterIds(): EntityId[] {
+		let submittedUserIds: EntityId[] = [];
+		const submittedSubmissions = this.getSubmittedSubmissions();
 
-		return count;
+		submittedSubmissions.forEach((submission) => {
+			const memberUserIds = submission.getSubmitterIds();
+			submittedUserIds = [...submittedUserIds, ...memberUserIds];
+		});
+
+		const uniqueSubmittedUserIds = [...new Set(submittedUserIds)];
+
+		return uniqueSubmittedUserIds;
 	}
 
-	// attention based on this parent use this.getParent() instant
-	getMaxSubmissions(): number {
-		// hack until parents are defined
-		const numberOfStudents = this.course ? this.course.getNumberOfStudents() : 0;
+	private getUniqueSubmitterWithGrade(): EntityId[] {
```

Comment:
Same question as above: Is it possible that they are not unique? 

And should be named "getUniqueSubmitterWithGradeIds".

Reply:
Already changed the comment before.